.clang-tidy

```diff
@@ -0,0 +1,70 @@
+---
+Checks: 'clang-analyzer-*,
+concurrency-*,
+cppcoreguidelines-*,
+misc-*,
+-misc-non-private-member-variables-in-classes,
+modernize-*,
```

Comment:
Just FYI, modernize-* also includes trailing return type. I have been using that recently, but it's up to you whether you want to commit to it on this repo.

Reply:
Yeah, after reviewing the flag it does appear interesting though I think for this project it's unneeded.